File: app/src/main/java/com/bccommittee/fragment/DashboardFragment.java
package com.bccommittee.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bccommittee.R;
import com.bccommittee.activity.AddMemberActivity;
import com.bccommittee.adapter.PaymentAdapter;
import com.bccommittee.database.DatabaseHelper;
import com.bccommittee.model.Fund;
import com.bccommittee.model.MonthlyReport;
import com.bccommittee.model.Payment;
import com.bccommittee.util.CurrencyUtil;

import java.util.List;

public class DashboardFragment extends Fragment {

    private long fundId;
    private DatabaseHelper db;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_dashboard, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        if (getArguments() != null) fundId = getArguments().getLong("fund_id", -1);
        db = DatabaseHelper.getInstance(requireContext());

        Fund fund = db.getFundById(fundId);
        if (fund == null) return;

        bindStats(view, fund);
        bindRecentPayments(view);
        bindOverdueAlert(view, fund);
        bindQuickActions(view, fund);
    }

    private void bindStats(View v, Fund fund) {
        List<MonthlyReport> reports = db.getReportsByFund(fundId);

        long pool = reports.isEmpty() ? 0 : reports.get(0).getTotalFund();
        long interestMonth = reports.isEmpty() ? 0 : reports.get(0).getInterestIn();

        ((TextView) v.findViewById(R.id.tv_total_pool))
            .setText(CurrencyUtil.format(pool));
        ((TextView) v.findViewById(R.id.tv_this_month))
            .setText(CurrencyUtil.format(fund.getMonthlyAmount()));
        ((TextView) v.findViewById(R.id.tv_interest_month))
            .setText(CurrencyUtil.format(interestMonth));
    }

    private void bindRecentPayments(View v) {
        RecyclerView rv = v.findViewById(R.id.rv_recent_payments);
        rv.setLayoutManager(new LinearLayoutManager(requireContext()));
        rv.setNestedScrollingEnabled(false);

        PaymentAdapter adapter = new PaymentAdapter(requireContext());
        rv.setAdapter(adapter);

        List<Payment> payments = db.getPaymentsByFund(fundId);
        // Show only first 5
        adapter.setPayments(payments.subList(0, Math.min(5, payments.size())));
    }

    private void bindOverdueAlert(View v, Fund fund) {
        LinearLayout alertLayout = v.findViewById(R.id.layout_overdue_alert);
        if (fund.isOverdue() && fund.getOverdueCount() > 0) {
            alertLayout.setVisibility(View.VISIBLE);
            ((TextView) v.findViewById(R.id.tv_overdue_title))
                .setText(fund.getOverdueCount() + " payment(s) overdue");
        } else {
            alertLayout.setVisibility(View.GONE);
        }
    }

    private void bindQuickActions(View v, Fund fund) {
        v.findViewById(R.id.btn_collect_payment).setOnClickListener(btn -> {
            // Navigate to Payments tab (index 2)
            if (getActivity() != null) {
                ((com.google.android.material.tabs.TabLayout)
                    getActivity().findViewById(R.id.tab_layout))
                    .selectTab(((com.google.android.material.tabs.TabLayout)
                        getActivity().findViewById(R.id.tab_layout)).getTabAt(2));
            }
        });

        v.findViewById(R.id.btn_give_loan).setOnClickListener(btn -> {
            if (getActivity() != null) {
                ((com.google.android.material.tabs.TabLayout)
                    getActivity().findViewById(R.id.tab_layout))
                    .selectTab(((com.google.android.material.tabs.TabLayout)
                        getActivity().findViewById(R.id.tab_layout)).getTabAt(1));
            }
        });

        v.findViewById(R.id.btn_add_member).setOnClickListener(btn -> {
            Intent intent = new Intent(requireContext(), AddMemberActivity.class);
            intent.putExtra("fund_id", fundId);
            intent.putExtra("fund_name", fund.getName());
            startActivity(intent);
        });
    }
}

File: app/src/main/kotlin/com/khaga/ksociety/fragment/DashboardFragment.kt
package com.khaga.ksociety.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.khaga.ksociety.R
import com.khaga.ksociety.activity.MainActivity
import com.khaga.ksociety.adapter.PaymentAdapter
import com.khaga.ksociety.databinding.FragmentDashboardBinding
import com.khaga.ksociety.util.CurrencyUtil
import com.khaga.ksociety.viewmodel.FundDetailViewModel

class DashboardFragment : Fragment() {

    private var _binding: FragmentDashboardBinding? = null
    private val binding get() = _binding!!
    private lateinit var viewModel: FundDetailViewModel
    private lateinit var recentAdapter: PaymentAdapter
    private var fundId = -1L

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentDashboardBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        fundId    = arguments?.getLong("fund_id") ?: -1L
        viewModel = ViewModelProvider(requireParentFragment())[FundDetailViewModel::class.java]

        recentAdapter = PaymentAdapter()
        binding.rvRecentPayments.apply {
            layoutManager = LinearLayoutManager(requireContext())
            adapter       = recentAdapter
            isNestedScrollingEnabled = false
        }

        setupObservers()
        setupQuickActions()
    }

    private fun setupObservers() {
        viewModel.fund.observe(viewLifecycleOwner) { fund ->
            _binding ?: return@observe
            fund ?: return@observe
            binding.tvThisMonth.text = CurrencyUtil.format(fund.monthlyAmount)
            binding.layoutOverdueAlert.visibility =
                if (fund.isOverdue && fund.overdueCount > 0) View.VISIBLE else View.GONE
            if (fund.isOverdue)
                binding.tvOverdueTitle.text = "${fund.overdueCount} payment(s) overdue"
        }
        viewModel.reports.observe(viewLifecycleOwner) { reports ->
            _binding ?: return@observe
            val latest = reports.maxByOrNull { it.id }
            binding.tvTotalPool.text     = CurrencyUtil.format(latest?.totalFund  ?: 0L)
            binding.tvInterestMonth.text = CurrencyUtil.format(latest?.interestIn ?: 0L)
        }
        viewModel.payments.observe(viewLifecycleOwner) { payments ->
            _binding ?: return@observe
            recentAdapter.submitList(payments.take(5))
        }
    }

    private fun setupQuickActions() {
        // Collect Payment — opens the 3-step PaymentFlowBottomSheet directly
        binding.btnCollectPayment.setOnClickListener {
            openPaymentFlow()
        }

        // Give Loan — open GiveLoanBottomSheet directly
        binding.btnGiveLoan.setOnClickListener {
            openGiveLoanSheet()
        }

        // Add Member — navigate to AddMemberFragment
        binding.btnAddMember.setOnClickListener {
            val f = AddMemberFragment().apply {
                arguments = Bundle().apply { putLong("fund_id", fundId) }
            }
            (requireActivity() as MainActivity).navigateTo(f, "add_member")
            (requireActivity() as MainActivity).hideBottomNav()
        }
    }

    private fun openPaymentFlow() {
        val members = viewModel.members.value.orEmpty()
        if (members.isEmpty()) {
            android.widget.Toast.makeText(
                requireContext(), "No members yet. Add members first.", android.widget.Toast.LENGTH_SHORT
            ).show()
            return
        }
        val sheet = PaymentFlowBottomSheet.newInstance(fundId)
        sheet.members = members
        sheet.onPaymentSaved = { payment ->
            viewModel.insertPayment(payment) { success ->
                activity?.runOnUiThread {
                    _binding ?: return@runOnUiThread
                    android.widget.Toast.makeText(
                        requireContext(),
                        if (success) "\u2705 \u20b9${payment.amount} saved!" else "Failed. Try again.",
                        android.widget.Toast.LENGTH_SHORT
                    ).show()
                }
            }
        }
        sheet.show(childFragmentManager, "payment_flow_dashboard")
    }

    private fun openGiveLoanSheet() {
        val members = viewModel.members.value.orEmpty()
        if (members.isEmpty()) {
            android.widget.Toast.makeText(
                requireContext(), "No members yet. Add members first.",
                android.widget.Toast.LENGTH_SHORT).show()
            return
        }
        val sheet = GiveLoanBottomSheet.newInstance()
        sheet.members = members
        sheet.onLoanIssued = { memberId, amount, _, _ ->
            val member = members.find { it.id == memberId }
            if (member != null) {
                val updated = member.copy(amtBorrowed = member.amtBorrowed + amount)
                viewModel.updateMember(updated) { success ->
                    activity?.runOnUiThread {
                        _binding ?: return@runOnUiThread
                        android.widget.Toast.makeText(requireContext(),
                            if (success) "\uD83D\uDCB8 Loan issued!" else "Failed.",
                            android.widget.Toast.LENGTH_SHORT).show()
                    }
                }
            }
        }
        sheet.show(childFragmentManager, "give_loan_dashboard")
    }

    override fun onDestroyView() { super.onDestroyView(); _binding = null }
}
